pdns/dnsdist-lua.cc

```diff
@@ -779,8 +779,8 @@ vector<std::function<void(void)>> setupLua(bool client, const std::string& confi
       return std::shared_ptr<DNSRule>(new SuffixMatchNodeRule(smn, quiet ? *quiet : false));
     });
 
-  g_lua.writeFunction("NetmaskGroupRule", [](const NetmaskGroup& nmg) {
-      return std::shared_ptr<DNSRule>(new NetmaskGroupRule(nmg));
+  g_lua.writeFunction("NetmaskGroupRule", [](const NetmaskGroup& nmg, bool src = true) {
```

Comment:
With the Lua wrapper we use, you need to make `src` a `boost::optional<bool>` if you want it to be optional, which is a good idea to keep compatibility. You'll then have to check whether it is set before using it. There is an example just above in `SuffixMatchNodeRule`.